tensorflow/tools/ci_build/linux/rocm/rocm_ci_sanity.sh

```diff
@@ -0,0 +1,547 @@
+#!/usr/bin/env bash
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+#
+# Usage: ci_sanity.sh [--pep8] [--incremental] [bazel flags]
+#
+# Options:
+#           run sanity checks: python 2&3 pylint checks and bazel nobuild
+#  --pep8   run pep8 test only
+#  --incremental  Performs checks incrementally, by using the files changed in
+#                 the latest commit
+
+# Current script directory
+SCRIPT_DIR=$( cd ${0%/*} && pwd -P )
+
+ROOT_DIR=$( cd "$SCRIPT_DIR/../../../../.." && pwd -P )
+source "${ROOT_DIR}/tensorflow/tools/ci_build/builds/builds_common.sh"
+
+# Helper functions
+die() {
+  echo $@
+  exit 1
+}
+
+num_cpus() {
+  # Get the number of CPUs
+  if [[ -f /proc/cpuinfo ]]; then
+    N_CPUS=$(grep -c ^processor /proc/cpuinfo)
+  else
+    # Fallback method
+    N_CPUS=`getconf _NPROCESSORS_ONLN`
+  fi
+  if [[ -z ${N_CPUS} ]]; then
+    die "ERROR: Unable to determine the number of CPUs"
+  fi
+
+  echo ${N_CPUS}
+}
+
+# Helper functions for examining changed files in the last non-merge git
+# commit.
+
+# Get the hash of the last non-merge git commit on the current branch.
+# Usage: get_last_non_merge_git_commit
+get_last_non_merge_git_commit() {
+  git rev-list --no-merges -n 1 HEAD
+}
+
+# List files changed (i.e., added, removed or revised) in the last non-merge
+# git commit.
+# Usage: get_changed_files_in_last_non_merge_git_commit
+get_changed_files_in_last_non_merge_git_commit() {
+  git diff-tree --no-commit-id --name-only -r $(get_last_non_merge_git_commit)
+}
+
+# List changed files for this branch relative to the parent branch.
+# The parent branch is assumed to be origin/develop-upstream.
```

Comment:
this assumption has two problems.
1. as already mentioned this will work only for PRs filed on the `develop-upstream` branch. this is the case for the vast majority of PRs, so is OK. but we to be cognizant this and remember not to enable this CI job for non develop-upstream PRs
1. What happens, if there are commits to the develop-upstream branch that came in after the branch for PR was forked? This should be OK too, assuming those commits were made after running the CI jobs on them (and hence should have no errors). Think there is a way to find the "base commit" for two branches (i.e. the commit at which the PR branch was forked from the `develop-upstream` branch) Using that commit to diff instead of `origin/develop-upstream` might be better.